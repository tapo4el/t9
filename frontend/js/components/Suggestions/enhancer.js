import { connect } from 'react-redux';

import { addWord } from '../../actions';
import { getSuggestions } from '../../selectors';

const mapStateToProps = state => ({
    suggestions: getSuggestions(state),
});

const mapDispatchToProps = dispatch => ({
    addWord: payload => dispatch(addWord(payload)),
});

export default connect(mapStateToProps, mapDispatchToProps);
