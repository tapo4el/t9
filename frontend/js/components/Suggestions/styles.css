.t9-app .suggestions {
    flex:1 1;
}
.t9-app .suggestions li {
    padding: 0 0 0 10px;
    cursor: pointer;
    line-height: 25px;
    font-size: 14px;
    font-weight: 100;
    border-radius: 5px;
    margin-right: 25px;
}
.t9-app .suggestions li:hover {
    background: #e0e1e2;
}
