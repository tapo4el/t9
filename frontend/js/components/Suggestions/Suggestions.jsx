import React from 'react';
import { func, arrayOf, string } from 'prop-types';

import './styles.css';

class Suggestions extends React.PureComponent {
    static propTypes = {
        suggestions: arrayOf(string),
        addWord: func.isRequired,
    };

    render() {
        const { suggestions, addWord } = this.props;
        return (
            <ul className="suggestions">
                {suggestions.map(val => <li key={val} onClick={() => addWord(val)}>{val}</li>)}
            </ul>
        )
    }
}

export default Suggestions;
