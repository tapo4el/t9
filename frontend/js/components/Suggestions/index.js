import Suggestions from './Suggestions';
import enhancer from './enhancer';

export default enhancer(Suggestions);
