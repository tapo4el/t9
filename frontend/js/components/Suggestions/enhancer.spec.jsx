import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import Suggestions from './index';

describe('Input enhancer', () => {
    const initialState = {
        suggestions: ['Hello', 'world'],
    };
    const mockStore = configureStore();
    const store = mockStore(initialState);
    const wrapper = mount(<Provider store={store}><Suggestions /></Provider>);

    it('should render correctly', () => {
        const component = renderer.create(
            <Suggestions store={store} />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('addWord should work correctly', () => {
        wrapper.find('li').at(0).simulate('click');
        const actions = store.getActions();

        expect(actions[0]).toEqual({ type: 'ADD_WORD', payload: 'Hello' });
    });
});
