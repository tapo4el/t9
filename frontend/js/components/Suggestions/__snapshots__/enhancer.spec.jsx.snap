// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input enhancer should render correctly 1`] = `
<ul
  className="suggestions"
>
  <li
    onClick={[Function]}
  >
    Hello
  </li>
  <li
    onClick={[Function]}
  >
    world
  </li>
</ul>
`;
